app/modules/intelligence/agents/custom_agents/custom_agents_service.py

```
@@ -1,68 +1,465 @@
+
+    def _convert_to_agent_schema(self, custom_agent: CustomAgentModel) -> Agent:
+        task_schemas = []
+        for i, task in enumerate(custom_agent.tasks, start=1):
+            expected_output = task.get("expected_output", {})
+            if isinstance(expected_output, str):
+                try:
+                    expected_output = json.loads(expected_output)
+                except json.JSONDecodeError:
+                    raise ValueError(
+                        "Invalid JSON format for expected_output in a task."
+                    )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)